### DISCUSSION TITLE
Release v0.16.0 Beta

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#2051] [Announcements]
## Features
- Complete database overhaul. Data should now much better sync up between windows (theme changes are still not synced)
- Comments now show pinned, owner, and reply from owner indicators
- Adding support for default custom arguments for external players
- FreeTube should now recognize more channel links when pasting through the search bar / sent from browser extensions
- You can now search through your history and playlists for videos
- You can now import / export your playlists. Please export your playlists at some point before the next release as the next minor release will likely be full playlists. Best to have a backup before that just in case
- Downloading videos / audio is now done within FreeTube. You will be notified when the download starts and when the download is finished
- Implemented "Stats for nerds" to show certain details about the video being watched. Right click on the video player to view it
- Middle clicking on a video will now open that video in a new window (Also prevents non-YouTube links from opening up in a separate window)

## Fixes
- Updated the ytdl-core module, which fixes the very recent buffering issue with videos from the local API
- Fixed an issue where dates were not translated properly
- Fixed various places where text would overflow from its container (ex.: autoplay, comments)
- Fixed an issue where a user's playlist page would scroll to the top when loading more videos
- The current playback speed will always be displayed when hovering over the list of playback rates

## Changes
- Electron has been updated to v16.0.8, fixing an issue with video playback on older versions of Windows
- The dislike count and the like / dislike ratio bar has been hidden due to the recent changes with YouTube. We'll look into possible alternates when a privacy respecting option exists
- Certain settings will automatically get disabled when incompatible settings are enabled
- Hovering over watched videos now show a smooth transition

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.16.0-beta'>Release v0.16.0 Beta</a>.</em>

**Comment by anonaddict** [comment]
Playlist fixes/and supporting private playlists, potentially even imports would be a gamechanger! Congratulations on another release

**Comment by ssnor** [comment]
Great job to everyone involved!

I usually update as soon as I see an interesting commit (yay Actions), and it has been a joy to see this project grow and become more and more feature complete.

Thank you for this project PrestoN, it's a great program!

**Comment by Privacy-Now** [comment]
What type of file is required to import a playlist? I'm trying to use DB Browser for SQLite to turn my NewPipe data into a CSV file. If that's correct them I'm certainly doing something wrong as it's not working.

**Reply by PrestonN** [reply]
FreeTube uses it's own `.db` extension, but it's read and configured as a `.json` file. It likely won't be the exact same format as what comes from NewPipe so it may need a few changes in order for FreeTube to accept it.

You can add a few videos to your favorites and then export that file to see what an example looks like.

**Reply by Privacy-Now** [reply]
Good to know. Thank you. I will continue to play around with it.

**Reply by anonaddict** [reply]
> FreeTube uses it's own .db extension, but it's read and configured as a .json file. It likely won't be the exact same format as what comes from NewPipe so it may need a few changes in order for FreeTube to accept it

Will there be support for .csv files? I have like 100 playlists from before I deleted my youtube account and I'd love to be able to import them

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
If i recall correctly importing subscriptions only supports csv files at the moment. There definitely will be support for importing csv files for playlists (once the implementation of playlists has been completed) and importing history from csv
